roc/version_test.go

```diff
@@ -80,3 +81,199 @@ func TestVersion_Parse(t *testing.T) {
 	}
 }
+
+func TestVersion_Validite(t *testing.T) {
+	tests := []struct {
+		name    string
+		version Versions
+		wantErr error
+	}{
+		{
+			name: "compatible: equal versions",
+			version: Versions{
+				Native: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+				Bindings: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+			},
+			wantErr: nil,
+		},
+		{
+			name: "incompatible: binding's major version less than native",
+			version: Versions{
+				Native: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+				Bindings: SemanticVersion{
+					Major: 0,
+					Minor: 1,
+					Patch: 1,
+				},
+			},
+			wantErr: errors.New(
+				"Detected incompatibility between roc bindings ( 0.1.1 ) and native library ( 1.1.1 ):" +
+					" Major versions are different",
+			),
+		},
+		{
+			name: "incompatible: binding's major greater than native",
+			version: Versions{
+				Native: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+				Bindings: SemanticVersion{
+					Major: 2,
+					Minor: 1,
+					Patch: 1,
+				},
+			},
+			wantErr: errors.New(
+				"Detected incompatibility between roc bindings ( 0.1.1 ) and native library ( 1.1.1 ):" +
+					" Major versions are different",
+			),
+		},
+		{
+			name: "compatible: binding's minor version greater than native",
+			version: Versions{
+				Native: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+				Bindings: SemanticVersion{
+					Major: 1,
+					Minor: 2,
+					Patch: 1,
+				},
+			},
+			wantErr: nil,
+		},
+		{
+			name: "incompatible: binding's minor version less than native",
+			version: Versions{
+				Native: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+				Bindings: SemanticVersion{
+					Major: 1,
+					Minor: 0,
+					Patch: 1,
+				},
+			},
+			wantErr: errors.New(
+				"Detected incompatibility between roc bindings ( 1.0.1 ) and native library ( 1.1.1 ):" +
+					" Minor version of bindings is less than native library",
+			),
+		},
+		{
+			name: "compatible: binding's patch less than native",
+			version: Versions{
+				Native: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 1,
+				},
+				Bindings: SemanticVersion{
+					Major: 1,
+					Minor: 1,
+					Patch: 0,
+				},
+			},
+			wantErr: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		err := tt.version.Validate()
+		if tt.wantErr != nil {
+			require.Error(t, err, tt.wantErr)
+		} else {
+			require.Nil(t, err)
+		}
+	}
+}
+
+func TestVersion_Check(t *testing.T) {
+	require.NotPanics(t, func() { versionCheck() })
+}
+
+var versionCheckCalled = false
+
+func TestVersion_Entrypoints(t *testing.T) {
+	versionCheckFn = func() {
+		versionCheckCalled = true
+	}
+
+	tests := []struct {
+		name       string
+		entrypoint func()
+	}{
+		{
+			name: "OpenContext",
+			entrypoint: func() {
+				ctx, _ := OpenContext(ContextConfig{})
+				defer ctx.Close()
+			}, // throws err
```

Comment:
What err?

Reply:
my bad had that comment when i didnt resolve the race condition error